tests/checkpoint/test_v1_checkpoint.py

```diff
@@ -0,0 +1,335 @@
+
+        assert cp.validation_definitions[1].name == validation_config_name_2
+        assert cp.validation_definitions[1].batch_definition.name == batch_config_name_2
+        assert cp.validation_definitions[1].suite.name == suite_name_2
+
+        # Check that all validation_definitions and nested suites have been assigned IDs during serialization
+        self._assert_valid_uuid(id=cp.validation_definitions[0].id)
+        self._assert_valid_uuid(id=cp.validation_definitions[1].id)
+        self._assert_valid_uuid(id=cp.validation_definitions[0].suite.id)
+        self._assert_valid_uuid(id=cp.validation_definitions[1].suite.id)
+
+    def _assert_valid_uuid(self, id: str | None) -> None:
+        if not id:
+            pytest.fail(
+                "id is None when it should be a valid UUID generated from a Store."
+            )
+
+        try:
+            uuid.UUID(id)
+        except ValueError:
+            pytest.fail(f"{id} is not a valid UUID.")
+
+    @pytest.mark.parametrize(
+        "serialized_checkpoint, expected_error",
+        [
+            pytest.param(
+                {
+                    "name": "my_checkpoint",
+                    "validation_definitions": [],
+                    "actions": [],
+                    "id": "c758816-64c8-46cb-8f7e-03c12cea1d67",
+                },
+                "Checkpoint must contain at least one validation definition",
+                id="missing_validations",
+            ),
+            pytest.param(
+                {
+                    "name": "my_checkpoint",
+                    "validation_definitions": [
+                        {
+                            "name": "i_do_not_exist",
+                            "id": "a758816-64c8-46cb-8f7e-03c12cea1d67",
+                        }
+                    ],
+                    "actions": [],
+                    "id": "c758816-64c8-46cb-8f7e-03c12cea1d67",
+                },
+                "Unable to retrieve validation definition",
+                id="nonexistent_validation",
+            ),
+            pytest.param(
+                {
+                    "name": "my_checkpoint",
+                    "validation_definitions": [
+                        {
+                            "other_key": "i_do_not_exist",
+                            "id": "a758816-64c8-46cb-8f7e-03c12cea1d67",
+                        }
+                    ],
+                    "actions": [],
+                    "id": "c758816-64c8-46cb-8f7e-03c12cea1d67",
+                },
+                "validation_definitions -> name\n  field required",
+                id="invalid_validation",
+            ),
+        ],
+    )
+    @pytest.mark.unit
+    def test_checkpoint_deserialization_failure(
+        self, serialized_checkpoint: str, expected_error: str
+    ):
+        with pytest.raises(ValidationError) as e:
+            Checkpoint.parse_obj(serialized_checkpoint)
```

Comment:
What's the difference between `parse_raw` used above and `parse_obj` used here? I was expected the same deserialization call in both places.

Reply:
`parse_raw` is on a string and `parse_obj` is on a dict